pkg/opensearch/response_parser_test.go

```diff
@@ -0,0 +1,1030 @@
+					   "key": 2000
+					 }
+				   ]
+				 }
+			   }
+			 }
+		   ]
+				}`
+		rp, err := newResponseParserForTest(targets, response)
+		assert.Nil(t, err)
+		result, err := rp.getTimeSeries()
+		assert.Nil(t, err)
+		assert.Len(t, result.Responses, 1)
+
+		queryRes := result.Responses["A"]
+		assert.NotNil(t, queryRes)
+		assert.Len(t, queryRes.Frames, 2)
+		seriesOne := queryRes.Frames[0]
+		assert.Equal(t, "p75", seriesOne.Name)
+		assert.Len(t, seriesOne.Fields, 2)
+		assert.Equal(t, time.Date(1970, time.January, 1, 0, 0, 1, 0, time.UTC), *seriesOne.Fields[0].At(0).(*time.Time))
+		assert.Equal(t, time.Date(1970, time.January, 1, 0, 0, 2, 0, time.UTC), *seriesOne.Fields[0].At(1).(*time.Time))
+		assert.EqualValues(t, 3.3, *seriesOne.Fields[1].At(0).(*float64))
+		assert.EqualValues(t, 2.3, *seriesOne.Fields[1].At(1).(*float64))
+
+		seriesTwo := queryRes.Frames[1]
+		assert.Equal(t, "p90", seriesTwo.Name)
+		assert.Len(t, seriesTwo.Fields, 2)
+		assert.Equal(t, time.Date(1970, time.January, 1, 0, 0, 1, 0, time.UTC), *seriesTwo.Fields[0].At(0).(*time.Time))
+		assert.Equal(t, time.Date(1970, time.January, 1, 0, 0, 2, 0, time.UTC), *seriesTwo.Fields[0].At(1).(*time.Time))
+		assert.EqualValues(t, 5.5, *seriesTwo.Fields[1].At(0).(*float64))
+		assert.EqualValues(t, 4.5, *seriesTwo.Fields[1].At(1).(*float64))
+	})
+
+	t.Run("With extended stats", func(t *testing.T) {
+		targets := map[string]string{
+			"A": `{
+					"timeField": "@timestamp",
+					"metrics": [{ "type": "extended_stats", "meta": { "max": true, "std_deviation_bounds_upper": true, "std_deviation_bounds_lower": true }, "id": "1" }],
+		 "bucketAggs": [
+						{ "type": "terms", "field": "host", "id": "3" },
+						{ "type": "date_histogram", "field": "@timestamp", "id": "4" }
+					]
+				}`,
+		}
+		response := `{
+		   "responses": [
+			 {
+			   "aggregations": {
+				 "3": {
+				   "buckets": [
+					 {
+					   "key": "server1",
+					   "4": {
+						 "buckets": [
+						   {
+							 "1": {
+							   "max": 10.2,
+							   "min": 5.5,
+							   "std_deviation_bounds": { "upper": 3, "lower": -2 }
+							 },
+							 "doc_count": 10,
+							 "key": 1000
+						   }
+						 ]
+					   }
+					 },
+					 {
+					   "key": "server2",
+					   "4": {
+						 "buckets": [
+						   {
+							 "1": {
+							   "max": 15.5,
+							   "min": 3.4,
+							   "std_deviation_bounds": { "upper": 4, "lower": -1 }
+							 },
+							 "doc_count": 10,
+							 "key": 1000
+						   }
+						 ]
+					   }
+					 }
+				   ]
+				 }
+			   }
+			 }
+		   ]
+				}`
+		rp, err := newResponseParserForTest(targets, response)
+		assert.Nil(t, err)
+		result, err := rp.getTimeSeries()
+		assert.Nil(t, err)
+		assert.Len(t, result.Responses, 1)
+
+		queryRes := result.Responses["A"]
+		assert.NotNil(t, queryRes)
+		assert.Len(t, queryRes.Frames, 6)
+
+		seriesOne := queryRes.Frames[0]
+		assert.Equal(t, "server1 Max", seriesOne.Name)
+		assert.Len(t, seriesOne.Fields, 2)
```

Comment:
nit: I think that the intent of the original was checking the number of data points rather than the number of fields on this, so maybe a line like `assert.Len(t, seriesOne.Fields[0], 1)` should be used?

Reply:
Nah, Fields is really at the same level that Points was and Points's length was all they were checking. 

Something that happened was organization was a little bit switched between Points and Fields.
For example Points[0][0] was a float64 and Points[0][1] was a time.
Now in Fields that would be: Fields[0][0] is time and Fields [1][0] is float64. So in the case where you only are describing one "data point", you would have had a Points of length 1, but now you would have a Fields of length 2.

However before they never checked the second dimension length. (They would have had a panic if they were then accessing a second dimension that didn't exist like Points[0][<doesn't exist>].) 
So I changed the following:
* changed any length assertions before accessing indexes to "require" so that the test doesn't proceed and print panics
* added the length requirement assertions to the second dimensions